packages/sanity/src/_internal/cli/actions/manifest/extractManifestAction.ts

```diff
@@ -0,0 +1,152 @@
+import {createHash} from 'node:crypto'
+import {mkdir, writeFile} from 'node:fs/promises'
+import {dirname, join, resolve} from 'node:path'
+import {Worker} from 'node:worker_threads'
+
+import {type CliCommandArguments, type CliCommandContext} from '@sanity/cli'
+import readPkgUp from 'read-pkg-up'
+
+import {
+  type CreateManifest,
+  type CreateWorkspaceManifest,
+  type ManifestWorkspaceFile,
+} from '../../../manifest/manifestTypes'
+import {type ExtractManifestWorkerData} from '../../threads/extractManifest'
+import {getTimer} from '../../util/timing'
+
+const MANIFEST_FILENAME = 'create-manifest.json'
+const SCHEMA_FILENAME_SUFFIX = '.create-schema.json'
+
+/** Escape-hatch env flags to change action behavior */
+const EXTRACT_MANIFEST_DISABLED = process.env.SANITY_CLI_EXTRACT_MANIFEST_ENABLED === 'false'
+const EXTRACT_MANIFEST_LOG_ERRORS = process.env.SANITY_CLI_EXTRACT_MANIFEST_LOG_ERRORS === 'true'
+
+const CREATE_TIMER = 'create-manifest'
+
+interface ExtractFlags {
+  path?: string
+}
+
+/**
+ * This method will never throw
+ */
+export async function extractManifestSafe(
+  args: CliCommandArguments<ExtractFlags>,
+  context: CliCommandContext,
+): Promise<void> {
+  if (EXTRACT_MANIFEST_DISABLED) {
+    return
+  }
+
+  try {
+    await extractManifest(args, context)
```

Comment:
Could it be a good idea to add a timeout to this operation, at least for initial iterations before we understand how it might perform for various schemas in the wild? We could set a fairly lengthy value (1 or 2 minutes, perhaps) that would at least ensure we don't block folks.

Reply:
Good point, was considering it but was a lazy boi 😅

Reply:
Added a 2 minute timeout.

Also added a line about which env var to set if you run into an extract error - probably useful if folks run into the timeout. Adding 5 seconds to a built is one thing, adding 2 minutes quite another 😓